src/github.com/getlantern/flashlight/util/http.go

```diff
@@ -2,21 +2,138 @@ package util
 
 import (
 	"crypto/tls"
+	"errors"
 	"fmt"
 	"net"
 	"net/http"
 	"net/url"
+	"sync/atomic"
 	"time"
 
+	"github.com/getlantern/fronted"
 	"github.com/getlantern/golog"
 	"github.com/getlantern/keyman"
 	"github.com/getlantern/waitforserver"
 )
 
+const (
+	defaultAddr = "127.0.0.1:8787"
```

Comment:
We may not want to hardcode proxy address here, though it's also the behaviour of 2.0.6.